core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -475,6 +475,14 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
   @LibraryOperator(libraries = {MYSQL, ORACLE})
   public static final SqlFunction REGEXP_REPLACE = new SqlRegexpReplaceFunction();
 
+  /** The "REGEXP_CONTAINS(value, regexp)" function.
+   * Returns TRUE if value is a partial match for the regular expression, regexp. */
+  @LibraryOperator(libraries = {BIG_QUERY})
```

Comment:
BigQuery [uses re2][1] for their regex engine. Google has an [implementation for Java][2]. We should use that here, both as a matter of syntax conformance (since this is a BigQuery-specific function) and for the security reasons outlined in the RE2J readme. This could be a follow-up, but I don't think we should close the bug until that's done.

[1]: https://cloud.google.com/bigquery/docs/reference/standard-sql/string_functions#regexp_contains
[2]: https://github.com/google/re2j